### ISSUE TITLE
Pixel is looking for a new maintainer

### ISSUE THREAD

**Issue by faiface** [faiface/pixel#225]
Hey guys, thank you all for sticking around!

In the last few months, it has become increasingly more clear that I am no longer able to maintain this repository. Pixel is and always will be one of my favorite projects and my proudest achievements. It was my first (and biggest) open-source success and brought me immense joy and satisfaction. Despite this, my focus and interests have shifted over time to other things and I can no longer gather enough motivation to sustainably maintain the project.

That's why we're looking for a new maintainer. It should be someone, who:
1. Is a good programmer and is experienced with the Go programming language.
2. Has (or wants to have) a passion for Pixel.
3. Is willing to spend time responding to issues, merging pull requests, or perhaps sometimes proposing, designing and implementing new features.
4. Is willing to make new releases and announce them on Reddit, or other platforms.

In other words, someone who wants to take good care of Pixel ;)

If you think you could be this new maintainer, write a reply to this issue.

The new maintainer will become fully responsible for the repository, I will not interfere with their decisions, but I will always be available for discussions, questions, mostly regarding the general direction of the project. If the new maintainer decides to open a Github Sponsors profile, 100% of the money will go to the new maintainer.

Once a new maintainer is in place, Pixel will enter a new phase, hopefully, a very prosperous one :P

That's all, hope we find somebody soon enough!

**Comment by delp** [comment]
Hey there,

I'd be interested in doing this. I've really enjoyed using Pixel. I think it's by far the best game library in go.

Here's my credentials

1) I've used pixel a fair amount
2) I work as an SE and own some Go code at my real job
3) I like solving problems and am passionate about open source projects and Go generally.

Ideally I could share the responsibility with someone but I'll do my best! Let me know what you think.

**Comment by faiface** [comment]
@delp Great, didn't expect a response so soon :D. Your credentials look good! I'll leave this open for at least two days to give other people a chance to respond as well. After that, we'll figure it out. Thanks!

**Comment by delp** [comment]
This is my first time maintaining a project of this size but I'll do my best. I would welcome anyone who wanted to be a co-maintainer.

I've got a lot of time on my hands at the moment due to being quarantined though. :)

**Comment by thegtproject** [comment]
FYI, whoever takes over Pixel will have to adopt me as well. As a diagnosed ADHD person, my attention span sucks but sometimes I get really motivated with adding features to Pixel :)

**Comment by faiface** [comment]
@thegtproject Haha, that's right Brandon. Thank you for all you've done for Pixel!

**Comment by delp** [comment]
I've got ADHD as well, so maybe we can form into a super maintainer by doing the DragonballZ fusion dance.

**Comment by delp** [comment]
Who would use a discord server if I set one up? Reply with thumbs up

(For discussing pull requests, asking for help, chatting about issues, showing off projects etc)

**Comment by faiface** [comment]
@delp Righ now, there is a Gitter server set up here: https://gitter.im/pixellib/Lobby

Discord has been something we've talked about a lot, but never implemented.

**Comment by Asday** [comment]
I would personally prefer that PRs remain on github.  Discord will die just like Skype did before it, and MSN messenger did before that.

Asking for help sounds fine out of band, but it would be nice if answers were turned into examples and put in the examples repo, or this repo's wiki.

**Comment by faiface** [comment]
I agree PRs and issues should remain here on GitHub, but replacing Gitter with Discord for casual chat, helping beginners, and so on could be a good idea.

**Comment by delp** [comment]
I don't want the long-form comment discussion to move, that is an important part of documenting changes. But I would like a place where I can chat in real time with folks about certain things.

**Comment by steelx** [comment]
discord please

**Comment by delp** [comment]
Alright, I set a server up, I'll post an invite.

OK, folks who contribute regularly... tell me if I'm thinking about the right kinds of things here:

* Should we try to improve unit test coverage? (batch.go and color.go have low coverage for example)
* I see there are 3 open PRs. What's the process like for testing changes? Just `go test`?

**Comment by svera** [comment]
My two cents:
* Improving unit test coverage never hurts, could be a nice first task to get to grips with the project.
* Regarding semantic versioning, I strongly agree. I would also try to do more frequent, smaller releases, 0.8 dates back to 2018.

**Comment by faiface** [comment]
Alright, @delp, it's time to make you the new maintainer :tada: :tada: :tada:

There's a lot of things that need to be dealt with:
- Lost of issues
- 3 open PRs
- The new Discord channel

Among the PRs, [this one is the most important](https://github.com/faiface/pixel/pull/211). Upgrading to GLFW 3.3  is a big step that needs to be done.

Feel free to contact me privately on Discord. I trust you will lead the project well, but I'm sure there will be some rough edges, things to figure out at least at the beginning, so I encourage you to discuss those things with me.

Regarding your questions:
- Improving unit test coverage is always welcome. Initially, Pixel was pretty poor with tests, mostly because I was lazy to do them. 100% test coverage is absolutely not a necessity, but if we have time to get close to it, that's good.
- The process for testing changes is `go test`, yes, but more importantly, test it manually. Try running the examples, if you have a hunch, try breaking the library. I feel like manual testing is at least as important in gamedev as automated testing, if not more important.
- Semantic versioning, I thought we did semantic versioning? Not from the beginning, but recently we did adopt it. Am I wrong?

**Comment by delp** [comment]
Discord Invite:
https://discord.gg/8v8BcW

**Comment by jamierocks** [comment]
I'd have gone for a timeless IRC channel, but Discord works too 👍

**Comment by pentaphobe** [comment]
Given orgs are free now, I thought I'd create one - however it seems both `go-pixel` (reg'd March 2020 - maybe by one of you?) and `pixelgl` are taken.  (Obviously plain old `pixel` is long gone)

Anyway - throwing my hat in the ring as another ADHD volunteer - happy to help however I can, albeit intermittently

Any other ideas for org name?

**Comment by pentaphobe** [comment]
> I'd have gone for a timeless IRC channel, but Discord works too 👍

nothing like a comms app that takes up more resources than the games running in the background :)

**Comment by delp** [comment]
The debian version has a segfault in it right now so I have to run it in a for loop.

**Comment by delp** [comment]
Let's be fancy and call it Pixel Special Interest Group like it was a kubernetes project.

I'm just kidding. I don't have any ideas.

I'm very excited to see that folks are still invested in this project so join the discord, and let's make things cool.

I started writing some unit tests today. 

Is it time to close this issue? It's becoming like a blog now. Sorry.

**Comment by pentaphobe** [comment]
@delp seems like the scope of the issue is covered, yeah

Aside: Alas the Discord invite has expired, any chance of an updated one?

**Comment by delp** [comment]
@pentaphobe  I've updated the README with a link that shouldn't expire.

Vote we leave this issue open for a few more days in the hope of perhaps a co-maintainer expressing interest.

**Comment by pentaphobe** [comment]
I'm down to co-maintain at least until someone preferable comes along

Have been using Pixel for a few months, and Go for a couple years - also I'm using Pixel for a side project so likely I'll be forking and raising some PRs regardless of maintainer status :)

**Comment by fgrosse** [comment]
I'm happy to see there is a lot of motivation around the project :)

Thanks @faiface again for starting the project initially and also for passing it to the next maintainer instead of letting it die :+1:

**Comment by delp** [comment]
@pentaphobe Awesome, that's great news. I'm going to go ahead and close this then as I feel like I've gotten a lot of community involvement.

**Comment by Seb-C** [comment]
@faiface @delp @thegtproject Are you guys still maintaining Pixel, or is it dead?

**Comment by Seb-C** [comment]
@dusk125 too 🙂
